### PR TITLE
Remove a smattering of (potential) allocations with spans

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Review by GrabYourPitchforks — APPROVED** [review]

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs [hunk 1]
```diff
@@ -541,7 +541,7 @@ private HashSet<int> ListUsedPrefixes(Dictionary<string, string> nsList, string
                         }
                         if (numeric)
                         {
-                            long index = long.Parse(name.Substring(prefixLength), NumberStyles.Integer, CultureInfo.InvariantCulture);
+                            long index = long.AsSpan(name.Substring(prefixLength), NumberStyles.Integer, CultureInfo.InvariantCulture);
```
Comment (bjornen77) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs]: Change to: `long index = long.Parse(name.AsSpan(` ?

Comment (stephentoub) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs]: Yes.  This was a typo.

**Comment by danmosemsft** [comment]
Please help add (exactly one) `area-XXX` label when the bot does not add one - that will help train it.

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs (hunk 1 above)
Comment (bjornen77) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs]: Except from that small typo, LGTM.

**Pull request merged**